CMakeLists.txt

```diff
@@ -2,22 +2,48 @@
 cmake_minimum_required(VERSION 3.4)
 project(libuv LANGUAGES C)
 
+cmake_policy(SET CMP0057 NEW)
+cmake_policy(SET CMP0063 NEW)
+cmake_policy(SET CMP0064 NEW)
+
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
+
 include(CMakePackageConfigHelpers)
 include(CMakeDependentOption)
+include(CheckCCompilerFlag)
 include(GNUInstallDirs)
 include(CTest)
 
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS ON)
+set(CMAKE_C_STANDARD 90)
+
 cmake_dependent_option(LIBUV_BUILD_TESTS
   "Build the unit tests when BUILD_TESTING is enabled and we are the root project" ON
   "BUILD_TESTING;CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR" OFF)
 
+cmake_dependent_option(LIBUV_BUILD_BENCH
+  "Build the benchmarks when building unit tests and we are the root project" ON
+  "LIBUV_BUILD_TESTS" OFF)
+
+check_c_compiler_flag(-Wstrict-prototypes UV_LINT_STRICT_PROTOTYPES)
+check_c_compiler_flag(-Wextra UV_LINT_EXTRA)
+check_c_compiler_flag(-Wall UV_LINT_DEFAULT)
+
+check_c_compiler_flag(-Wno-unused-parameter UV_LINT_NO_UNUSED_PARAMETER)
+
+set(lint-strict-prototypes $<$<BOOL:${UV_LINT_STRICT_PROTOTYPES}>:-Wstrict-prototypes>)
+set(lint-default $<$<BOOL:${UV_LINT_STRICT_DEFAULT}>:-Wall>)
+set(lint-extra $<$<BOOL:${UV_LINT_STRICT_EXTRA}>:-Wextra>)
```

Comment:
Is this The Right Way (TM) to do it now? I get the compiler agnostic argument, but it feels much more verbose: it's a 3 step process.

Reply:
It's currently three steps, however I plan to put a lot of this boilerplate into a "UV.cmake" (or some other named file) so we can backport a few CMake bug fixes and make this sort of thing appear to be a one-line call.

Reply:
Me likey!

Reply:
Still on the TODO, but it will be coming :)